File: lib/src/main/java/me/ameriod/lib/mvp/view/deligate/ViewGroupMvpDelegateCallback.java
package me.ameriod.lib.mvp.view.deligate;

import android.os.Parcelable;

import me.ameriod.lib.mvp.Mvp;

public interface ViewGroupMvpDelegateCallback<V extends Mvp.View, P extends Mvp.Presenter<V>>
        extends MvpDelegateCallback<V, P> {

    Parcelable superOnSaveInstanceState();

    void superOnRestoreInstanceState(Parcelable state);

}

File: lib/src/main/java/me/ameriod/lib/mvp/deligate/ViewGroupMvpDelegateCallback.kt
package me.ameriod.lib.mvp.deligate

import android.os.Parcelable

import me.ameriod.lib.mvp.Mvp

interface ViewGroupMvpDelegateCallback<out V : Mvp.View, out P : Mvp.Presenter<out V>> : MvpDelegateCallback<V, P> {

    fun superOnSaveInstanceState(): Parcelable

    fun superOnRestoreInstanceState(state: Parcelable)
}
